### PR TITLE
Architectural endpoint changes

### PR DESCRIPTION
**Pull request by dride-for-sure**
Related to: #78

### CODE REVIEW

**Review by jamarob — COMMENTED** [review]
Nice! I think the renaming helped to make your intentions clearer. 🏫 

I am a little bit confused by your route naming, I would expect to find all routes related to authentication below `auth` and the api routes simply below `api`. We can configure access permissions, so there is really no need to namespace the routes.

File: backend/src/main/java/org/opensource/schulaltas/controller/model/SchoolDto.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
  private String name;
  private Address address;
  private Contact contact;
+ private String type;
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/controller/model/SchoolDto.java:24]: Thought of an enum too. In the future I would write the different types in the database to be able to manage them via the cms. In this respect I have already chosen no enum for this.

File: backend/src/main/java/org/opensource/schulaltas/service/SchoolService.java [hunk 2]
```diff
@@ -30,7 +31,21 @@ public SchoolService (SchoolDb schoolDb, GeoService geoService, TimeUTC timeUTC,
...
 
- public Optional<School> getSchool (String number) {
+ public List<TypeDto> listTypes () {
+  List<String> types = schoolDb.findAll().stream()
+                               .map( school -> school.getType() )
+                               .collect( Collectors.toList() );
+  List<String> uniqueTypes = new ArrayList<>( new HashSet<>( types ) );
+  return uniqueTypes.stream()
+                 .map( type -> TypeDto.builder().name( type ).count( Collections.frequency( types, type ) ).build() )
+                 .collect( Collectors.toList() );
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/service/SchoolService.java:41]: Well...youre right! The list wrapper is useless. Have been digging through the docu yesterday evening and overlooked `distinct`. Thanks nice!

File: backend/src/main/java/org/opensource/schulaltas/controller/PrivatePropertyController.java [hunk 3]
```diff
@@ -9,7 +9,7 @@
 import java.util.List;
 
 @RestController
-@RequestMapping ("/auth/property")
+@RequestMapping ("/auth/v1/property")
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/controller/PrivatePropertyController.java:12]: I could migrate to

`/api` for all public routes
`/admin` for all private routes
`/auth` for login purpose only

Suggestions?

File: backend/src/main/java/org/opensource/schulaltas/service/SchoolService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/service/SchoolService.java:41]: Ich weiß jetzt, warum ich `distinct` gestern verworfen habe, denn es kürzt leider kaum / garnicht ab:

```java
public List<TypeDto> listTypes () {
  List<String> types = schoolDb.findAll().stream()
                               .map( school -> school.getType() )
                               .collect( Collectors.toList() );
  List<String> uniqueTypes = new ArrayList<>( new HashSet<>( types ) );
  return uniqueTypes.stream()
                 .map( type -> TypeDto.builder().name( type ).count( Collections.frequency( types, type ) ).build() )
                 .collect( Collectors.toList() );
 }
```
wird zu:
```java
public List<TypeDto> listTypes () {
  List<String> types = schoolDb.findAll().stream()
                               .map( school -> school.getType() )
                               .collect( Collectors.toList() );
  List<String> uniqueTypes = types.stream().distinct().collect( Collectors.toList() );
  return uniqueTypes.stream()
                 .map( type -> TypeDto.builder().name( type ).count( Collections.frequency( types, type ) ).build() )
                 .collect( Collectors.toList() );
 }
```
Möglicherweise hast du noch eine Idee das zu vereinfachen?

[the remark this answers is not in the document]
Comment (jamarob) [reply backend/src/main/java/org/opensource/schulaltas/service/SchoolService.java:41]:
```Java
return schoolDb.findAll().stream()
    .map( school -> school.getType() )
    .distinct()
    .map( type -> TypeDto.builder().name( type ).count( Collections.frequency( types, type ) ).build() )
    .collect( Collectors.toList() );
```
I think this is short and sweet enough. Maybe extract the lambda of the second map to a helper function more brevity?

[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/service/SchoolService.java:41]: _--> reference to slack conversation_ ... sum up: no abbreviate version possible :(

**Comment by dride-for-sure** [comment]
Thanks for the review @jamarob! I appreciate it. :pray:

File: backend/src/main/java/org/opensource/schulaltas/controller/PrivatePropertyController.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamarob) [reply backend/src/main/java/org/opensource/schulaltas/controller/PrivatePropertyController.java:12]: sounds good to me ;)

**Pull request merged**